### PR TITLE
refactor SDC Newton solver to use burn_t

### PR DESCRIPTION
**Pull request by zingale**
this cleans up some interfaces and reduces memory requirements.

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
test suite diffs are roundoff:
http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2023-10-05-005/index.html

**Comment by zingale** [comment]
this is ready for review.  I will follow it up with more cleaning, but I wanted to get these changes in first.

**Comment by zingale** [comment]
okay, test suite passes now with no diffs:
http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2023-10-05-006/index.html

File: Source/sdc/Castro_sdc_util.H [hunk 1]
```diff
@@ -170,9 +168,13 @@ sdc_update_o2(const int i, const int j, const int k,
 
         sdc_solve(dt_m, U_old, U_new, C_zone, sdc_iteration);
 
-        // we solved our system to some tolerance, but let's be sure we are conservative by
-        // reevaluating the reactions and { doing the full step update
-        single_zone_react_source(U_new, R_full, burn_state);
+        // we solved our system to some tolerance, but let's be sure
+        // we are conservative by reevaluating the reactions and {
```
Comment (maxpkatz) [inline Source/sdc/Castro_sdc_util.H:172]: superfluous {

Comment (zingale) [reply Source/sdc/Castro_sdc_util.H:172]: why?
The idea is that we might have done it one a little bit at a time, and next we want to do it in terms of averages and as a single update

**Review by maxpkatz — APPROVED** [review]

File: Source/sdc/Castro_sdc_util.H (hunk 1 above)
Comment (zingale) [reply Source/sdc/Castro_sdc_util.H:172]: oh... I see what you mean ;)

**Pull request merged**